### PR TITLE
Fix serverless tests for new eventing

### PR DESCRIPTION
**Pull request by grischperl**
**Description**

Changes proposed in this pull request:

- Fix serverless tests for new eventing

The chart changes are in PR https://github.com/kyma-project/kyma/pull/10793 where serverless tests are passing in the gke-integration jobs.

### CODE REVIEW

File: tests/function-controller/pkg/subscription/subscription.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+		Jitter:   backOffJitter,
+	}
+	return retry.OnError(customBackoff, func(err error) bool {
+		if err != nil {
+			return true
+		}
+		return false
+	}, func() error {
+		ctx := context.Background()
+		subUnstructured, err := s.resCli.ResCli.Get(ctx, subscription.Name, metav1.GetOptions{})
```
Comment (grischperl) [inline tests/function-controller/pkg/subscription/subscription.go:140]: Could the Get() method be used, or is it too much as it has another retry in itself?

File: tests/function-controller/pkg/subscription/subscription.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+func (s *Subscription) Get() (*unstructured.Unstructured, error) {
+	subscription, err := s.resCli.Get(s.name)
+	if err != nil {
+		return &unstructured.Unstructured{}, errors.Wrapf(err, "while getting Subscription %s in namespace %s", s.name, s.namespace)
+	}
+
+	return subscription, nil
+}
+
+func (s *Subscription) WaitForStatusRunning(subscription *kymaeventingv1alpha1.Subscription) error {
```
Comment (grischperl) [inline tests/function-controller/pkg/subscription/subscription.go:126]: There is no need anymore to pass the subscription to the function. For the subscription name s.Name could be used?

Comment (sayanh) [reply tests/function-controller/pkg/subscription/subscription.go]: Right, changed

File: tests/function-controller/pkg/subscription/subscription.go (hunk 1 above)
Comment (sayanh) [reply tests/function-controller/pkg/subscription/subscription.go]: The `Get()` is from `tests/function-controller/vendor/k8s.io/client-go/dynamic/simple.go` which doesn't have a retry inside. Moreover, this is Get() is critical to check the latest status change in the resource. Why do you think it has another retry?

Comment (grischperl) [reply tests/function-controller/pkg/subscription/subscription.go]: I mean the subscription.Get, which calls the Get from here: `tests/function-controller/pkg/resource/resource.go`

Comment (sayanh) [reply tests/function-controller/pkg/subscription/subscription.go]: Changed to get() from subscription.

File: tests/function-controller/cmd/main.go [hunk 3]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/kyma-project/kyma/tests/function-controller/pkg/step"
 	"github.com/kyma-project/kyma/tests/function-controller/testsuite"
 	"github.com/kyma-project/kyma/tests/function-controller/testsuite/scenarios"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```
[the remark this answers is not in the document]
Comment (sayanh) [reply tests/function-controller/cmd/main.go:18]: I'd leave it here as it enables local development.

**Review by sayanh — APPROVED** [review]

File: tests/function-controller/pkg/subscription/subscription.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+		log:         c.Log,
+		verbose:     c.Verbose,
+	}
+}
+
+func (s *Subscription) Create(sinkUrl *url.URL) (subscription *kymaeventingv1alpha1.Subscription, err error) {
+	eventSource := &kymaeventingv1alpha1.Filter{
+		Type:     "exact",
+		Property: "source",
+		Value:    "",
```
Comment (m00g3n) [inline tests/function-controller/pkg/subscription/subscription.go:62]: Is it equivalent of a wildcard?

**Review by m00g3n — COMMENTED** [review]
A couple of questions

File: tests/function-controller/pkg/subscription/subscription.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Subscription",
+			APIVersion: "eventing.kyma-project.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.name,
+			Namespace: s.namespace,
+		},
+		Spec: kymaeventingv1alpha1.SubscriptionSpec{
+			Protocol:         "",
```
[the remark this answers is not in the document]
Comment (sayanh) [reply tests/function-controller/pkg/subscription/subscription.go:89]: No, it should be blank for NATS now. The controller is forgiving. We will bring strict validation for this later.

File: tests/function-controller/pkg/subscription/subscription.go (hunk 4 above)
Comment (sayanh) [reply tests/function-controller/pkg/subscription/subscription.go:62]: Not really, for NATS this should be blank for now. The source(or application name in Kyma context) is part of event-type for now. This source will be re-evaluated when BEB comes into picture.

File: tests/function-controller/cmd/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (sayanh) [reply tests/function-controller/cmd/main.go:18]: Yes, as I said before it enables local development and in terms of security, it is safe as compared to azure. We should resolve it holistically throughout the project and start using build tags. Anyway, I got rid of it.

**Review by sayanh — APPROVED** [review]

**Review by m00g3n — APPROVED** [review]

File: tests/function-controller/go.mod [hunk 6]
```diff
@@ -10,24 +10,29 @@ require (
...
-	k8s.io/apimachinery v0.18.12
+	k8s.io/api v0.19.3
+	k8s.io/apimachinery v0.19.3
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
 	sigs.k8s.io/controller-runtime v0.6.4
 )
 
 replace (
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.3.1
+	github.com/kyma-project/kyma => github.com/kyma-project/kyma v0.5.1-0.20210302111801-5a4f7610cf45
```
Comment (dbadura) [inline tests/function-controller/go.mod:30]: Do we need this?

Comment (sayanh) [reply tests/function-controller/go.mod:30]: I think it is needed as the latest gnostic was not compliant with the rest.

Comment (dbadura) [reply tests/function-controller/go.mod:30]: I mean `kyma-project`

**Pull request merged**